airflow-core/src/airflow/jobs/triggerer_job_runner.py

```diff
@@ -244,6 +271,14 @@ def upload_to_remote(self):
         upload_to_remote(self.bound_logger)
 
 
+@functools.cache
+def in_process_api_server() -> InProcessExecutionAPI:
```

Comment:
Do we need to cache this and the client property of the Supervisor? Wouldn't it be enough to only cache the resulting client object?

(Yes, you are probably copying my own PR, so in effect I'm critiquing my own code here)

Reply:
Oh you'r right we probably don't. Caching the client should be enough

Reply:
I will update that at both places

Reply:
Done in https://github.com/apache/airflow/pull/48239/commits/1848af980fe00870d30b8d540f5502cbc4a95cab